libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/RouteRefresh.java

```diff
@@ -75,11 +77,23 @@ public void refresh(RouteProgress routeProgress, RefreshCallback refreshCallback
   }
 
   private void refresh(final DirectionsRoute directionsRoute, final int legIndex, RefreshCallback refreshCallback) {
+    if (isInvalid(directionsRoute, refreshCallback)) {
+      return;
+    }
     MapboxDirectionsRefresh.builder()
       .requestId(directionsRoute.routeOptions().requestUuid())
       .routeIndex(Integer.valueOf(directionsRoute.routeIndex()))
       .legIndex(legIndex)
       .accessToken(accessToken)
       .build().enqueueCall(new RouteRefreshCallback(directionsRoute, legIndex, refreshCallback));
   }
+
+  private boolean isInvalid(DirectionsRoute directionsRoute, RefreshCallback refreshCallback) {
+    String requestUuid = directionsRoute.routeOptions().requestUuid();
+    if (TextUtils.isEmpty(requestUuid) || directionsRoute.routeIndex() == null) {
+      refreshCallback.onError(new RefreshError("RouteProgress passed has invalid DirectionsRoute"));
```

Comment:
NIT Could we extract this magic number into a constant?

Reply:
Not sure if we should report this as an "error" until we know why the Refresh API is returning `null` after some time stale. Should we maybe stop the periodic refresh for this scenario? 🤔

Reply:
I think it's fine to send the error, our behavior is to fail silently:
```
  @Override
  public void onError(RefreshError error) {
    Timber.w(error.getMessage());
    routeRefresher.updateIsChecking(false);
  }
```
I extracted the magic number 👍